asf_heat_pump_suitability/analysis/hn_zones/comparison_of_hn_zones.py

```diff
@@ -0,0 +1,350 @@
+        pl.col("DESNZ_pilot_score") == pilot_score
+    )
+    avg_hn_score_pilot = (
+        filtered_la_hp_suitability_scores.select(
+            pl.col("HN_N_avg_score_weighted").mean()
+        ).item()
+        if filtered_la_hp_suitability_scores.height > 0
+        else 0.0
+    )
+    return avg_hn_score_pilot
+
+
+def calculate_mae_for_pilot_score(
+    hp_suitability_scores_with_desnz: pl.DataFrame, score_value: int
+) -> float:
+    """
+    Calculate and log the Mean Absolute Error (MAE) for entries with a specified 'DESNZ_pilot_score'.
+
+    Args:
+        hp_suitability_scores_with_desnz (pl.DataFrame): DataFrame containing 'DESNZ_pilot_score' and 'absolute_error' columns.
+        score_value (int): The value of 'DESNZ_pilot_score' (must be 1 or 0) for which to calculate the MAE.
+
+    Returns:
+        float: The MAE for the specified 'DESNZ_pilot_score'.
+
+    Raises:
+        ValueError: If score_value is not 1 or 0.
+    """
+    # Validate that score_value is either 1 or 0
+    if score_value not in {0, 1}:
+        raise ValueError("score_value must be either 1 or 0")
+
+    mae = hp_suitability_scores_with_desnz.filter(
+        pl.col("DESNZ_pilot_score") == score_value
+    )["absolute_error"].mean()
+
+    logging.info(
+        f"Mean Absolute Error (MAE) for DESNZ_pilot_score = {score_value}: {mae}"
+    )
+    return mae
+
+
+def calculate_mae_for_all(
+    hp_suitability_scores: pl.DataFrame, desnz_col: str, nesta_hn_score_col: str
+) -> Tuple[pl.DataFrame, float]:
+    """
+    Calculate the Mean Absolute Error (MAE) between the DESNZ pilot score and Nesta's HN suitability score, add the absolute error column to the DataFrame,
+    and log the result.
+
+    Args:
+        hp_suitability_scores_with_desnz (pl.DataFrame): DataFrame containing the DESNZ pilot (actual) and Nesta HN suitability score (predicted) columns.
+        desnz_col (str): Name of the column with DESNZ pilot (actual) values.
+        nesta_hn_score_col (str): Name of the column with Nesta HN suitability (predicted) values.
+
+    Returns:
+        Tuple[pl.DataFrame, float]:
+            - Updated DataFrame with 'absolute_error' column added.
+            - The Mean Absolute Error (MAE) between the actual and predicted columns.
+    """
+    # Calculate absolute error and add it as a new column
+    hp_suitability_scores_with_error = hp_suitability_scores.with_columns(
+        (pl.col(desnz_col) - pl.col(nesta_hn_score_col)).abs().alias("absolute_error")
+    )
+
+    # Calculate the mean of the absolute error
+    mae = hp_suitability_scores_with_error["absolute_error"].mean()
+    logging.info(
+        f"Mean Absolute Error (MAE) for {desnz_col} vs {nesta_hn_score_col}: {mae}"
+    )
+
+    return hp_suitability_scores_with_error, mae
+
+
+def main():
+    # Define the output directory and set up logging
+    output_dir = os.path.join(
+        PROJECT_DIR, "asf_heat_pump_suitability/analysis/hn_zones/output_data/"
```

Comment:
I would suggest moving the hn_zone dir to `outputs/hn_zones` and then just save the results directly into that dir :)

Reply:
Ah ok, I misunderstood, I thought you meant moving the whole directory to the outputs folder. I will move it over after we've finished this review :).

Reply:
After weighing it up in my head, I do think it probably makes sense to have a version saved to s3 (so people can play around with the plotting scripts without needing to run the comparison script) but if you are iterating locally, then make sure you save the files locally because it takes ages otherwise hehe and is a bit annoying.